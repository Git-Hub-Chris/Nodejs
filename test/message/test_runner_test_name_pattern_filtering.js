// Flags: --no-warnings --test-name-pattern=suite-one>suite-two>subtest-one
'use strict';
const common = require('../common');
const {
  after,
  afterEach,
  before,
  beforeEach,
  describe,
  it,
  test,
} = require('node:test');

describe('suite-one', () => {
  before(common.mustCall());
  beforeEach(common.mustCall(3));
  afterEach(common.mustCall(3));
  after(common.mustCall());

  it('subtest-one', common.mustNotCall());
  describe(
    'suite-two',
    common.mustCall(() => {
      it('subtest-one', common.mustCall());
    }),
  );
  describe('suite-three', common.mustNotCall());
});

test(
  'suite-one',
  common.mustCall(async (t) => {
    t.beforeEach(common.mustCall(3));
    t.afterEach(common.mustCall(3));

    await t.test('subtest-one', common.mustNotCall());
    await t.test(
      'suite-two',
      common.mustCall(async (t) => {
        await t.test('subtest-one', common.mustCall());
      }),
    );
    await t.test('suite-three', common.mustNotCall());
  }),
);

describe('excluded', () => {
  before(common.mustNotCall());
  beforeEach(common.mustNotCall());
  afterEach(common.mustNotCall());
  after(common.mustNotCall());

  it('subtest-one', common.mustNotCall());
  describe('suite-two', common.mustNotCall());
});

test('excluded', (t) => {
  t.beforeEach(common.mustNotCall());
  t.afterEach(common.mustNotCall());

  t.test('subtest-one', common.mustNotCall());
  t.test('suite-two', common.mustNotCall());
});
