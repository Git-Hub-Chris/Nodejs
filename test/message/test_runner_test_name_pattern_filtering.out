TAP version 13
# Subtest: suite-one
    # Subtest: subtest-one
    ok 1 - subtest-one # SKIP test name does not match pattern
      ---
      duration_ms: *
      ...
    # Subtest: suite-two
        # Subtest: subtest-one
        ok 1 - subtest-one
          ---
          duration_ms: *
          ...
        1..1
    ok 2 - suite-two
      ---
      duration_ms: *
      ...
    # Subtest: suite-three
    ok 3 - suite-three # SKIP test name does not match pattern
      ---
      duration_ms: *
      ...
    1..3
ok 1 - suite-one
  ---
  duration_ms: *
  ...
# Subtest: suite-one
    # Subtest: subtest-one
    ok 1 - subtest-one # SKIP test name does not match pattern
      ---
      duration_ms: *
      ...
    # Subtest: suite-two
        # Subtest: subtest-one
        ok 1 - subtest-one
          ---
          duration_ms: *
          ...
        1..1
    ok 2 - suite-two
      ---
      duration_ms: *
      ...
    # Subtest: suite-three
    ok 3 - suite-three # SKIP test name does not match pattern
      ---
      duration_ms: *
      ...
    1..3
ok 2 - suite-one
  ---
  duration_ms: *
  ...
# Subtest: excluded
ok 3 - excluded # SKIP test name does not match pattern
  ---
  duration_ms: *
  ...
# Subtest: excluded
ok 4 - excluded # SKIP test name does not match pattern
  ---
  duration_ms: *
  ...
1..4
# tests 4
# pass 2
# fail 0
# cancelled 0
# skipped 2
# todo 0
# duration_ms *
